### PR TITLE
BO: Remove unused variable

### PR DESCRIPTION
**Pull request by idnovate**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | `$context` variable is not used in function
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | 
| How to test?      | 
| Possible impacts? | 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23374)

### CODE REVIEW

**Comment by idnovate** [comment]
I've searched all the coincidences for `Product::searchByName` and there was only 1 function that uses 4 arguments

File: classes/Product.php [hunk 1]
```diff
@@ -4764,12 +4764,8 @@ public static function cacheFrontFeatures($product_ids, $id_lang)
      *
      * @return array|false Matching products
      */
-    public static function searchByName($id_lang, $query, Context $context = null, $limit = null)
+    public static function searchByName($id_lang, $query, $limit = null)
```
Comment (matks) [inline classes/Product.php]: Hi @idnovate I agree with you that this variable, if unused, is completely useless ; however by removing it we introduce a risk.

Let's say we merge this PR in `develop` : that will be delivered in PrestaShop 1.7.8

If today a module uses `Product::searchByName($id_lang, $query, null, $limit);` with PS 1.7.7 it works well.
Tomorrow the shop upgrades from 1.7.7 to 1.7.8 and now `Product::searchByName($id_lang, $query, null, $limit);` has a very different meaning as `$limit` argument is now the 3rd arg, not the 4th.

So I think we can _deprecate_ this parameter by adding a phpDoc statement, or even a warning. But I dont think the value of removing this useless argument is higher than the cost of breaking modules relying on this function.

I think the BC break should be avoided here.

If however `develop` was meant to be released as a new major version, where BC breaks are allowed, that would be totally legit 😄  .

**Review by matks — CHANGES_REQUESTED** [review]

File: classes/Product.php (hunk 1 above)
Comment (idnovate) [reply classes/Product.php]: I agree with you. So I have just removed the `if` inside the function, but keep the arguments.

**Review by Progi1984 — DISMISSED** [review]

File: classes/Product.php [hunk 2]
```diff
@@ -4766,10 +4766,6 @@ public static function cacheFrontFeatures($product_ids, $id_lang)
      */
     public static function searchByName($id_lang, $query, Context $context = null, $limit = null)
```
Comment (matks) [inline classes/Product.php:4767]: May I suggest to also modify the phpDoc ?

```
* @param Context|null $context
```
⬇️ 
```
* @param Context|null $context Deprecated, obsolete parameter not used anymore
```

**Review by matks — DISMISSED** [review]
Thank you for the great work ! 😄

File: classes/Product.php [hunk 3]
```diff
@@ -4759,17 +4759,13 @@ public static function cacheFrontFeatures($product_ids, $id_lang)
...
      * @param int $id_lang Language identifier
      * @param string $query Search query
-     * @param Context|null $context
+     * @param Context|null $context Deprecated, obsolete parameter not used anymore
      * @param int|null $limit
      *
      * @return array|false Matching products
      */
     public static function searchByName($id_lang, $query, Context $context = null, $limit = null)
     {
```
Comment (Progi1984) [inline classes/Product.php:4768]:
```suggestion
    {
        if ($context !== null) {
            Tools::displayParameterAsDeprecated('context');
        }
```

**Review by Progi1984 — COMMENTED** [review]
A last change and it's perfect

**Review by Progi1984 — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Re-running CI

**Review by matthieu-rolland — COMMENTED** [review]
CI OK => ready for merge (?)

**Comment by matthieu-rolland** [comment]
thank you @idnovate 👍

**Pull request merged**